### PR TITLE
 Inform Jersey of broken pipe by throwing exception

### PR DESCRIPTION
**Pull request by spericas**
Ensure exceptions are propagated so that Jersey can detect problems writing to a broken pipe. This is very useful when the server streams data in chunks, and possibly over a long period, such as when streaming SSE events. A new test that verifies the fix has been added.

Signed-off-by: Santiago Pericas-Geertsen <santiago.pericasgeertsen@oracle.com>

### CODE REVIEW

File: common/reactive/src/main/java/io/helidon/common/reactive/OutputStreamPublisher.java [hunk 1]
```diff
@@ -144,6 +144,9 @@ private void publish(byte[] buffer, int offset, int length) throws IOException {
         } catch (ExecutionException e) {
             complete(e.getCause());
             throw new IOException(e.getCause());
+        } catch (IllegalStateException e) {
```
Comment (ljnelson) [inline common/reactive/src/main/java/io/helidon/common/reactive/OutputStreamPublisher.java]: One minor comment.  Clearly your changes to `onNext()` now throw `IllegalStateException`.  Does anything else in this "wide" `try` block also throw `IllegalStateException`?  If so, is it OK to handle all such `IllegalStateExceptions` here in this manner?  Or should this `IllegalStateException` part of the `catch` block be "scoped tighter" around just the `onNext` call?  I don't know the answer.  I am just raising the point.

Comment (spericas) [reply common/reactive/src/main/java/io/helidon/common/reactive/OutputStreamPublisher.java]: It's a good question. I haven't seen other places, but I will look a bit more closely.

Comment (spericas) [reply common/reactive/src/main/java/io/helidon/common/reactive/OutputStreamPublisher.java]: I've now verified that just `onNext` can throw that exception.

**Comment by spericas** [comment]
I have now removed the WIP label

File: common/reactive/src/main/java/io/helidon/common/reactive/BaseProcessor.java [hunk 2]
```diff
@@ -76,6 +76,8 @@ public final void onNext(T item) {
             } catch (Throwable ex) {
                 onError(ex);
             }
+        } else {
```
Comment (tomas-langer) [inline common/reactive/src/main/java/io/helidon/common/reactive/BaseProcessor.java]: I would consider changing the if statement to remove the negation, e.g. (pseudo):
```
if (sub.closed()) {
  throw ISE;
} else {
  hookOnNext
}
```

As analyzing if statements with negation and else branch is a brain twister

File: common/reactive/src/test/java/io/helidon/common/reactive/BaseProcessorTest.java [hunk 3]
```diff
@@ -80,7 +80,11 @@ public void testOnNextAfterOnComplete() {
         processor.subscribe(subscriber);
         subscriber.request1();
         processor.onComplete();
-        processor.onNext("foo");
+        try {
```
Comment (tomas-langer) [inline common/reactive/src/test/java/io/helidon/common/reactive/BaseProcessorTest.java]: You can use
```
import static org.junit.jupiter.api.Assertions.assertThrows;
//....
assertThrows(IllegalStateException.class, () -> processor.onNext("foo");
```

File: common/reactive/src/main/java/io/helidon/common/reactive/BaseProcessor.java (hunk 2 above)
Comment (spericas) [reply common/reactive/src/main/java/io/helidon/common/reactive/BaseProcessor.java]: Will change, don't have a preference either way.

File: common/reactive/src/test/java/io/helidon/common/reactive/BaseProcessorTest.java (hunk 3 above)
Comment (spericas) [reply common/reactive/src/test/java/io/helidon/common/reactive/BaseProcessorTest.java]: Sure, that would read better.

**Pull request merged**